app/code/Magento/Eav/Model/Entity/Attribute/Backend/AbstractBackend.php

```diff
@@ -203,12 +203,12 @@ public function getEntityValueId($entity)
     /**
      * Retrieve default value
      *
-     * @return mixed
+     * @return string
      */
     public function getDefaultValue()
     {
         if ($this->_defaultValue === null) {
-            if ($this->getAttribute()->getDefaultValue()) {
+            if ($this->getAttribute()->getDefaultValue() !== null) {
```

Comment:
Here you touch the logic that relates to all attributes. Are you sure that all other attributes have `null` as default value? Won't it break their work?

Reply:
Hi @slavvka,

I have two arguments for this point.
First: auto tests have passed successfully.
Second: if you look at the screenshot under, there are list of attributes with default value '0'. After smoke check I have not found strict check '===' with null or 0 values anymore (exception - 'links_exist' attribute).
So, I assume, there should not be any problems for existing set of attributes.
P.S. have checked on Open Source Magento.
![image](https://user-images.githubusercontent.com/9420807/46918232-6178ea80-cfd8-11e8-83e2-00a83a4b4305.png)